cfme/infrastructure/provider/__init__.py

```diff
@@ -24,70 +19,34 @@
-    AngularSelect, toolbar as tb, Radio, match_location, BootstrapSwitch)
-from cfme.web_ui.form_buttons import FormButton
-from cfme.web_ui.tabstrip import TabStripForm
-from utils import conf, version, deferred_verpick
+from cfme.web_ui import Quadicon, paginator, toolbar as tb, match_location
+from utils import conf, version
 from utils.appliance import Navigatable
 from utils.appliance.implementations.ui import navigator, CFMENavigateStep, navigate_to
 from utils.log import logger
+from utils.net import resolve_hostname
 from utils.pretty import Pretty
 from utils.varmeth import variable
 from utils.wait import wait_for
 
 match_page = partial(match_location, controller='ems_infra', title='Infrastructure Providers')
-
-properties_form = TabStripForm(
-    fields=[
-        ('type_select', AngularSelect("emstype")),
-        ('name_text', Input("name")),
-        ("api_version", AngularSelect("api_version")),
-    ],
-    tab_fields={
-        "Default": [
-            ('hostname_text', Input("default_hostname")),
-            ('api_port', Input("default_api_port")),
-            ('sec_protocol', AngularSelect("default_security_protocol", exact=True)),
-            ('verify_tls_switch', BootstrapSwitch(input_id="default_tls_verify")),
-            ('ca_certs', Input('default_tls_ca_certs')),
-        ],
-        "Events": [
-            ('event_selection', Radio('event_stream_selection')),
-            ('amqp_hostname_text', Input("amqp_hostname")),
-            ('amqp_api_port', Input("amqp_api_port")),
-            ('amqp_sec_protocol', AngularSelect("amqp_security_protocol", exact=True)),
-        ],
-        "C & U Database": [
-            ('candu_hostname_text', Input("metrics_hostname")),
-            ('acandu_api_port', Input("metrics_api_port")),
-        ]
-    })
-
-prop_region = Region(locators={'properties_form': properties_form})
-
-cfg_btn = partial(tb.select, 'Configuration')
 pol_btn = partial(tb.select, 'Policy')
-mon_btn = partial(tb.select, 'Monitoring')
 
 
-class InfraProvider(Pretty, CloudInfraProvider):
+class InfraProvider(Pretty, CloudInfraProvider, Fillable):
     """
     Abstract model of an infrastructure provider in cfme. See VMwareProvider or RHEVMProvider.
 
     Args:
         name: Name of the provider.
         details: a details record (see VMwareDetails, RHEVMDetails inner class).
-        credentials (:py:class:`Credential`): see Credential inner class.
         key: The CFME key of the provider in the yaml.
-        candu: C&U credentials if this is a RHEVMDetails class.
-
+        endpoints: one or several provider endpoints like DefaultEndpoint
     Usage:
 
         myprov = VMwareProvider(name='foo',
-                             region='us-west-1',
-                             credentials=Provider.Credential(principal='admin', secret='foobar'))
+                             region='us-west-1'
+                             endpoints=DefaultEndpoint())
```

Comment:
**REQUIRED:** We should also show how to add the credentials like in the example above.